samza-log4j2/src/main/java/org/apache/samza/logging/log4j2/StreamAppender.java

```diff
@@ -73,7 +73,7 @@ public class StreamAppender extends AbstractAppender {
   private final BlockingQueue<EncodedLogEvent> logQueue = new LinkedBlockingQueue<>(DEFAULT_QUEUE_SIZE);
 
   private SystemStream systemStream = null;
-  private SystemProducer systemProducer = null;
+  protected SystemProducer systemProducer = null;
```

Comment:
why does this have to be `protected` if `sendEventToSystemProducer` is exposed to child classes?

Reply:
+1

Reply:
good question. This is mainly for testing purposes. oss samza has MockSystemProducer to test StreamAppender https://github.com/apache/samza/blob/ea14805a0588ac3cd7ffbef0e23463962bf9246b/samza-log4j2/src/test/java/org/apache/samza/logging/log4j2/MockSystemProducer.java

However, LI samza's InLogsAppender inherits StreamAppender and it doesn't have a way to inject this dependency and mock its behavior.
In the long run, it could be beneficial to use composition instead of inheritance. It means a refactor of current LI samza code

Reply:
why not package private then instead?

Reply:
sry, I didn't follow. can you clarify which package you suggest that should be private?

Reply:
we want to mock a SystemProducerException to mock the case that a kafka exception thrown by SystemProducer in InLogsAppender. And InLogsAppender is not in the same package, we need to set systemProducer for InLogsAppender

Reply:
It seems like this needs to be revisited on why this isn't mock friendly for unit tests outside the package.
Let us get this for now.